### PR TITLE
Replace the publication fields with a production field

### PR DESCRIPTION
**Pull request by alexwlchan**
For #1784. Includes notes from chatting to Silver, and some skeleton bits to get everything working.

### CODE REVIEW

**Comment by alexwlchan** [comment]
Yes, this PR is a bit of a monster diff. 😭 

A lot of old files have been deleted, and there are several entirely new files – so it doesn’t all need close scrutiny, I hope!

File: catalogue_pipeline/transformer/src/main/scala/uk/ac/wellcome/platform/transformer/transformers/sierra/SierraPublishers.scala [hunk 1]
```diff
@@ -1,51 +0,0 @@
...
-
-trait SierraPublishers {
-
-  // Populate wwork:publishers.
-  //
-  // We use MARC field "260", if populated, otherwise we use field "264".
-  //
-  // In either case:
-  //  - "label" comes from subfield $b
-  //  - "type" is "Organisation".
```
Comment (alexwlchan) [inline catalogue_pipeline/transformer/src/main/scala/uk/ac/wellcome/platform/transformer/transformers/sierra/SierraPublishers.scala]: @silveroliver We were tying $b as an Organisation originally – according to the new rules, it’s just an Agent. Should we still use the more specific type?

**Comment by alexwlchan** [comment]
Travis has now timed out the matcher job **six times**. I am tired of restarting it, so I’m going to run it locally here.

File: catalogue_pipeline/transformer/src/main/scala/uk/ac/wellcome/platform/transformer/transformers/sierra/SierraProduction.scala [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    subfieldTag: String): List[Unidentifiable[Agent]] =
+    vf.subfields
+      .filter { _.tag == subfieldTag }
+      .map { sf: MarcSubfield =>
+        Agent(label = sf.content)
+      }
+      .map { ag: Agent =>
+        Unidentifiable(ag)
+      }
+
```
Comment (mklander) [inline catalogue_pipeline/transformer/src/main/scala/uk/ac/wellcome/platform/transformer/transformers/sierra/SierraProduction.scala]: can this be a single map to `Unidentifiable(Agent(label = sf.content))`?

File: ontologies/WIP/sierratransformable [hunk 3]
```diff
@@ -459,38 +459,20 @@
...
-   * Example of how we might pull out year or date when we know it in the future:
-   * 
-   *   "publicationDate": {
-   *     "label": "May 23 1932",
-   *     "type": "Period" ,
-   *     "year": "1932" ,
-   *     "date" : "05-23-1932"
-   *  },
-   */
+    // @@AWLC
```
Comment (alicefuzier) [inline ontologies/WIP/sierratransformable]: should this still be here?

File: sbt_common/display/src/test/scala/uk/ac/wellcome/display/models/v1/DisplayWorkV1Test.scala [hunk 4]
```diff
@@ -67,73 +68,6 @@ class DisplayWorkV1Test extends FunSpec with Matchers {
     displayWork.identifiers shouldBe Some(List())
   }
 
-  describe("publishers") {
-    it("parses a work without any publishers") {
```
Comment (alicefuzier) [inline sbt_common/display/src/test/scala/uk/ac/wellcome/display/models/v1/DisplayWorkV1Test.scala]: Why are these tests deleted? Isn't v1 supposed to stay as it was?

File: catalogue_pipeline/transformer/src/main/scala/uk/ac/wellcome/platform/transformer/transformers/sierra/SierraProduction.scala (hunk 2 above)
Comment (alexwlchan) [reply catalogue_pipeline/transformer/src/main/scala/uk/ac/wellcome/platform/transformer/transformers/sierra/SierraProduction.scala]: I thought I’d tried that and it didn’t work, but it seems happy enough now. Changed!

File: ontologies/WIP/sierratransformable (hunk 3 above)
Comment (alexwlchan) [reply ontologies/WIP/sierratransformable]: Mmm. This is a file of notes, the @@AWLC is just a note of who wrote them. I’ll leave it for now, as it’s useful context.

File: sbt_common/display/src/test/scala/uk/ac/wellcome/display/models/v1/DisplayWorkV1Test.scala (hunk 4 above)
Comment (alexwlchan) [reply sbt_common/display/src/test/scala/uk/ac/wellcome/display/models/v1/DisplayWorkV1Test.scala]: As discussed, V1 DisplayWorks should have no publishers!

**Comment by alexwlchan** [comment]
I’m going to merge this as-is, matcher failures be damned.

**Pull request merged**